### PR TITLE
Issues #137, #145, #147, #148, #149, and #151

### PR DESCRIPTION
**Pull request by texadactyl**
@telegraphic 
If you approve of this merge, you can remove the obsolete ipynb file from ```tests```.

### CODE REVIEW

**Comment by texadactyl** [comment]
DIST=py3_manual.docker worked as expected.
DIST=py3_kern.docker failed:
```E   ModuleNotFoundError: No module named 'numpy.testing.nosetester'```
An environmental error?

**Comment by texadactyl** [comment]
Same thing happened again in DIST=py3_kern.docker:

/usr/lib/python3/dist-packages/scipy/_lib/_numpy_compat.py:10: in <module>
    from numpy.testing.nosetester import import_nose
E   ModuleNotFoundError: No module named 'numpy.testing.nosetester'

DIST=py3_manual.docker worked as expected.

**Comment by telegraphic** [comment]
I think we need to update the dockerfiles, as KERN3 has been EOL'd (plus it uses Py3.5, which breaks a lot of newer Py3 code with f-strings).

**Comment by texadactyl** [comment]
@marksbrt @telegraphic 
You'll note that some of the calculations were removed simply because the results were never used.

**Comment by marksbrt** [comment]
Hi all, I haven't looked at this code in a long time, but the changes outlined above look good. Please let me know if you need anything else from me.

**Comment by texadactyl** [comment]
@telegraphic 
FYI, Python version in DIST=py3_kern.docker is 3.6.7

**Comment by texadactyl** [comment]
Once this is merged, please update pypi (issue #150).

**Comment by texadactyl** [comment]
At least, we can provide the real reason for failing to instantiate a Waterfall object when the file header has a value that cannot be currently supported (E.g. nbits=4).  Better than a traceback with highlighting an error that was actually downstream of the real issue.

File: blimpy/waterfall.py [hunk 1]
```diff
@@ -389,7 +392,7 @@ def cmd_tool(args=None):
     parser.add_argument('-S', action='store_true', default=False, dest='save_only',
                        help='Turn off plotting of data and only save to file.')
     parser.add_argument('-D', action='store_false', default=False, dest='blank_dc',
```
Comment (telegraphic) [inline blimpy/waterfall.py:394]: (reminder to fix this `action='store_false', default=False`)

**Review by telegraphic — APPROVED** [review]
A-OK, KERN test suite failure is separate issue which I'll fix after merging

**Pull request merged**